src/spyglass/utils/dj_mixin.py

```diff
@@ -716,27 +768,133 @@ def fetch1(self, *args, log_fetch=True, **kwargs):
+
+    def see_banned_tables(self):
+        """Print banned tables."""
+        logger.info(f"Banned tables: {self._banned_search_tables}")
+
+    def restrict_by(
+        self,
+        restriction: str = True,
+        direction: str = "up",
+        return_graph: bool = False,
+        verbose: bool = False,
+        **kwargs,
+    ) -> QueryExpression:
+        """Restrict self based on up/downstream table.
+
+        If fails to restrict table, the shortest path may not have been correct.
+        If there's a different path that should be taken, ban unwanted tables.
+
+        >>> my_table = MyTable() # must be instantced
+        >>> my_table.ban_search_table(UnwantedTable1)
+        >>> my_table.ban_search_table([UnwantedTable2, UnwantedTable3])
+        >>> my_table.unban_search_table(UnwantedTable3)
+        >>> my_table.see_banned_tables()
+        >>>
+        >>> my_table << my_restriction
+
+        Parameters
+        ----------
+        restriction : str
+            Restriction to apply to the some table up/downstream of self.
+        direction : str, optional
+            Direction to search for valid restriction. Default 'up'.
+        return_graph : bool, optional
+            If True, return FindKeyGraph object. Default False, returns
+            restricted version of present table.
+        verbose : bool, optional
+            If True, print verbose output. Default False.
+
+        Returns
+        -------
+        Union[QueryExpression, FindKeyGraph]
+            Restricted version of present table or FindKeyGraph object. If
+            return_graph, use all_ft attribute to see all tables in cascade.
+        """
+        from spyglass.utils.dj_graph import (
+            TableChain,
+            TableChains,
+        )  # noqa: F401
+
+        if restriction is True:
+            return self
+        try:
+            ret = self.restrict(restriction)  # Save time trying first
+            logger.warning("Restriction valid for this table. Using as is.")
+            return ret
+        except DataJointError:
+            pass  # Could avoid try/except if assert_join_compatible return bool
+            logger.debug("Restriction not valid. Attempting to cascade.")
+
+        if direction == "up":
+            parent, child = None, self
+        elif direction == "down":
+            parent, child = self, None
+        else:
+            raise ValueError("Direction must be 'up' or 'down'.")
+
+        graph = TableChain(
+            parent=parent,
+            child=child,
+            direction=direction,
+            search_restr=restriction,
+            banned_tables=self._banned_search_tables,
+            allow_merge=True,
+            cascade=True,
+            verbose=verbose,
+            **kwargs,
+        )
+
+        if return_graph:
+            return graph
+
+        ret = graph.leaf_ft[0]
+        if len(ret) == len(self) or len(ret) == 0:
```

Comment:
The stop condition would either be finding a non-empty set or hitting the error where the key is no longer in the graph

Reply:
My plan for the immediate future is to get a sense of how useful it is from folks. If useful, if the table banning process is a chore, I would continue the `TableChain` search until all tables in the path had a valid restriction. Rather than deciding the path, I would depth-first search until there was a valid restriction the whole way. 

It seemed like a lot of work to put in without that initial feedback, but I can revisit if you think it's needed

Reply:
That's fair.  I agree it's not required for this PR.  

As another option that doesn't require action right now; maybe a more efficient solution would be something like a `required_tables_list` where a chain is only valid if it contains the tables in that list. Would potentially solve the case of picking from a certain source of a merge table.